components/proxy/src/main/java/com/hotels/styx/admin/handlers/livenessreadiness/Readiness.java

```diff
@@ -0,0 +1,71 @@
+/*
+  Copyright (C) 2013-2019 Expedia Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.hotels.styx.admin.handlers.livenessreadiness;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableMap;
+import com.hotels.styx.api.HttpHandler;
+import com.hotels.styx.api.HttpResponse;
+import com.hotels.styx.common.http.handler.HttpMethodHandler;
+import com.hotels.styx.configstore.ConfigStore;
+
+import static com.google.common.net.MediaType.JSON_UTF_8;
+import static com.hotels.styx.api.HttpHeaderNames.CONTENT_TYPE;
+import static com.hotels.styx.api.HttpMethod.GET;
+import static com.hotels.styx.api.HttpResponse.response;
+import static com.hotels.styx.api.HttpResponseStatus.OK;
+import static com.hotels.styx.api.HttpResponseStatus.SERVICE_UNAVAILABLE;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * Provides an endpoint for checking that Styx is ready to serve traffic.
+ */
+public final class Readiness {
+    private static final HttpResponse LOADED_RESPONSE = response(OK)
+            .header(CONTENT_TYPE, JSON_UTF_8)
+            .body(jsonBody(true), UTF_8)
+            .build();
+
+    private static final HttpResponse NOT_LOADED_RESPONSE = response(SERVICE_UNAVAILABLE)
+            .header(CONTENT_TYPE, JSON_UTF_8)
+            .body(jsonBody(false), UTF_8)
+            .build();
+
+    private Readiness() {
+    }
+
+    public static HttpHandler readinessHandler(ConfigStore configStore) {
+        return new HttpMethodHandler.Builder()
+                .add(GET, request -> generateResponse(configStore))
+                .build();
+    }
+
+    private static HttpResponse generateResponse(ConfigStore configStore) {
+        boolean loaded = configStore.get("loading.styx", Object.class).isPresent();
```

Comment:
The attribute name `loading.styx` suggests that the server is still actively attempting to start up.

Why not to call it `server.state` with values of `loading` and `ready`? Also this is bit problematic because Styx can open multiple server ports: http and https. Presumably it will be ready when the both are open.